V3 feature audio - clean up

Drapersniper opened the pull request:
Fixes broken stuff from previous PRs 
Fixes #3467, where paths with special characters no longer return an empty list from `glob.glob`
Attempts to fix #3506 by ensuring that `[p]playlist update` no longer uses the local cache
When selecting a folder with child folders with `[p]localplay` the bot will not enqueue all nested tracks

Drapersniper committed:
Merge remote-tracking branch 'Red-DiscordBot-UpStream/V3/develop' into audio-refactor-1

Drapersniper committed:
Merge remote-tracking branch 'Red-DiscordBot-UpStream/V3/develop' into audio-refactor-1

Drapersniper committed:
Lets reforce travis

Signed-off-by: Draper <27962761+Drapersniper@users.noreply.github.com>

Drapersniper committed:
Lets reforce travis

Signed-off-by: Draper <27962761+Drapersniper@users.noreply.github.com>

aikaterna committed:
Merge pull request #3363 from Drapersniper/audio-refactor-1

Audio Refactor: Part 1  - APIs

aikaterna committed:
Merge pull request #3363 from Drapersniper/audio-refactor-1

Audio Refactor: Part 1  - APIs

Drapersniper committed:
Merge remote-tracking branch 'Red-DiscordBot-UpStream/V3/feature/audio' into audio-refactor-2

Drapersniper committed:
Merge remote-tracking branch 'Red-DiscordBot-UpStream/V3/feature/audio' into audio-refactor-2

aikaterna committed:
Merge pull request #3424 from Drapersniper/audio-refactor-2

Audio refactor part 2 - Cog Class

aikaterna committed:
Merge pull request #3424 from Drapersniper/audio-refactor-2

Audio refactor part 2 - Cog Class

Commit:
Merge branch 'V3/feature/audio' of https://github.com/Cog-Creators/Red-DiscordBot into V3/feature/audio

Commit:
Merge branch 'V3/feature/audio' of https://github.com/Cog-Creators/Red-DiscordBot into V3/feature/audio

aikaterna committed:
Merge pull request #3500 from Drapersniper/V3/Feature/Audio/3

V3/feature/audio/3

aikaterna committed:
Merge pull request #3500 from Drapersniper/V3/Feature/Audio/3

V3/feature/audio/3

Drapersniper committed:
Merge remote-tracking branch 'Red-DiscordBot/V3/feature/audio' into V3/Feature/Audio/fixes/3485

Drapersniper committed:
Merge remote-tracking branch 'Red-DiscordBot/V3/feature/audio' into V3/Feature/Audio/fixes/3485

Drapersniper committed:
Merge branch 'V3/Feature/Audio/fixes/3485' of https://github.com/Drapersniper/Red-DiscordBot into V3/Feature/Audio/fixes/3485

Drapersniper committed:
Merge branch 'V3/Feature/Audio/fixes/3485' of https://github.com/Drapersniper/Red-DiscordBot into V3/Feature/Audio/fixes/3485

aikaterna committed:
Merge pull request #3502 from Drapersniper/v3/feature/audio/fixes/3485

V3/feature/audio/fixes/3485

aikaterna committed:
Merge pull request #3502 from Drapersniper/v3/feature/audio/fixes/3485

V3/feature/audio/fixes/3485

Drapersniper committed:
Merge remote-tracking branch 'Red-DiscordBot/V3/feature/audio' into V3/Feature/Audio/fixes/3353

Drapersniper committed:
Merge remote-tracking branch 'Red-DiscordBot/V3/feature/audio' into V3/Feature/Audio/fixes/3353

Drapersniper committed:
Merge branch 'V3/Feature/Audio/fixes/3353' of https://github.com/Drapersniper/Red-DiscordBot into V3/Feature/Audio/fixes/3353

Drapersniper committed:
Merge branch 'V3/Feature/Audio/fixes/3353' of https://github.com/Drapersniper/Red-DiscordBot into V3/Feature/Audio/fixes/3353

aikaterna committed:
Merge pull request #3503 from Drapersniper/v3/feature/audio/fixes/3353

V3/feature/audio/fixes/3353

aikaterna committed:
Merge pull request #3503 from Drapersniper/v3/feature/audio/fixes/3353

V3/feature/audio/fixes/3353

Drapersniper committed:
Fixes + Black

Improves Logging when `--debug` is used
Append Track Author to Track name if it is missing
Added Global Daily Playlists as part of the server daily playlists

Fixes `[p]playlist dedupe` not removing tracks
Fixes Error messages not using bot colour for embeds

Signed-off-by: Draper <27962761+Drapersniper@users.noreply.github.com>

Drapersniper committed:
Fixes + Black

Improves Logging when `--debug` is used
Append Track Author to Track name if it is missing
Added Global Daily Playlists as part of the server daily playlists

Fixes `[p]playlist dedupe` not removing tracks
Fixes Error messages not using bot colour for embeds

Signed-off-by: Draper <27962761+Drapersniper@users.noreply.github.com>

Drapersniper committed:
Add the guild object to the mocked context

Signed-off-by: Draper <27962761+Drapersniper@users.noreply.github.com>

Drapersniper committed:
Add the guild object to the mocked context

Signed-off-by: Draper <27962761+Drapersniper@users.noreply.github.com>

Drapersniper committed:
Add lyrics preference settable by server

Signed-off-by: Draper <27962761+Drapersniper@users.noreply.github.com>

Drapersniper committed:
Add lyrics preference settable by server

Signed-off-by: Draper <27962761+Drapersniper@users.noreply.github.com>

Drapersniper committed:
Fixes broken stuff from previous PRs
Fixes #3467, where paths with special characters no longer return an empty list from `glob.glob`
Attempts to fix #3506 by ensuring that `[p]playlist update` no longer uses the local cache
When selecting a folder with child folders with `[p]localplay` the bot will not enqueue all nested tracks

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fixes broken stuff from previous PRs
Fixes #3467, where paths with special characters no longer return an empty list from `glob.glob`
Attempts to fix #3506 by ensuring that `[p]playlist update` no longer uses the local cache
When selecting a folder with child folders with `[p]localplay` the bot will not enqueue all nested tracks

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fixes broken stuff from previous PRs
Fixes #3467, where paths with special characters no longer return an empty list from `glob.glob`
Attempts to fix #3506 by ensuring that `[p]playlist update` no longer uses the local cache
When selecting a folder with child folders with `[p]localplay` the bot will not enqueue all nested tracks

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fix Local track search string

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fix Local track search string

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fix Local track search string

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Add a start up task to the cog to is run in `def setup()`  to ensure the remaining tasks are not created prior to the cog being loaded

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Add a start up task to the cog to is run in `def setup()`  to ensure the remaining tasks are not created prior to the cog being loaded

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Add a start up task to the cog to is run in `def setup()`  to ensure the remaining tasks are not created prior to the cog being loaded

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Allow the bot to take the last spot in a VC in the room has a user limit set.

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Allow the bot to take the last spot in a VC in the room has a user limit set.

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Allow the bot to take the last spot in a VC in the room has a user limit set.

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Rename `userlimit` check to `is_vc_full` for further clarity

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Rename `userlimit` check to `is_vc_full` for further clarity

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Rename `userlimit` check to `is_vc_full` for further clarity

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

PredaaA commented:
After talks in Discord, I put this as a reminder for you @Drapersniper.
When something is blacklisted in `[p]audioset restrictions`, like Spotify, it should not do any fetch to Spotify API. Not only when trying to enqueue them, but also with `[p]playlist save | append` commands.

Another thing is that `Loading track xx/xx...` message on `[p]playlist save | append` commands get locked when the job is finished.

Drapersniper committed:
Address Jacks review of method names

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Address Jacks review of method names

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Address Jacks review of method names

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
update `[p]audioset cache` usage string

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
update `[p]audioset cache` usage string

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
update `[p]audioset cache` usage string

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Hey Aika ... I know I said no more features ... but this is technically a fix, to sort out spotify's shitty responses from their API matching really random shit ... this allows server to set the country code for its responses ATM this only impacts the `genre` command but in future can be used for other stuff (pls no bulli)

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Hey Aika ... I know I said no more features ... but this is technically a fix, to sort out spotify's shitty responses from their API matching really random shit ... this allows server to set the country code for its responses ATM this only impacts the `genre` command but in future can be used for other stuff (pls no bulli)

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Hey Aika ... I know I said no more features ... but this is technically a fix, to sort out spotify's shitty responses from their API matching really random shit ... this allows server to set the country code for its responses ATM this only impacts the `genre` command but in future can be used for other stuff (pls no bulli)

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
improve error handling

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
improve error handling

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
improve error handling

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper commented:
This is not ready to be merged, further work is required, and a full functionality review is required however once this one is proven to be functional
https://github.com/Cog-Creators/Red-DiscordBot/pull/3459 should be ready to be merged into V3/develop.

The "further work required" is ensuring all bugs currently in this branch are resolved.

The way i coded this branch, it contacts all previous work that went into the feature branch so everything can be tested here.

Once this is merged into the feature branch the audio refactor can be considered complete.

Drapersniper committed:
Merge branch 'C3-feature-audio-fixes' of https://github.com/Drapersniper/Red-DiscordBot into C3-feature-audio-fixes

Drapersniper committed:
Merge branch 'C3-feature-audio-fixes' of https://github.com/Drapersniper/Red-DiscordBot into C3-feature-audio-fixes

Drapersniper committed:
Merge #3644

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Style change

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Style change

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Merge remote-tracking branch 'Red-DiscordBot-UpStream/V3/feature/audio' into audio-refactor-2

Drapersniper committed:
Merge #3644

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Merge branch 'C3-feature-audio-fixes' of https://github.com/Drapersniper/Red-DiscordBot into C3-feature-audio-fixes

Drapersniper committed:
Remove Unused imports
Fix a few attribute references
stop built-in shadowing
Fix a few incorrect fingerprints

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fix playlist lazymatching

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

aikaterna reviewed (commented):
Not an exhaustive list. This is what I've found so far - I think you may have already addressed the playlist picker as I see from one of your later commits. 

[p]queue cleanself
Please title case "Removed tracks from the queue" like the rest of the error title messages.

[p]help Audio says `Class joining all Audio subclasses` - make this docstring more user friendly

[p]audioset autoplay reset - possibly display default value in message? (message now is `Set auto-play playlist to default value.`)

[p]audioset countrycode - make the docstring specify Spotify

No matching is present in playlist names across scopes.
Example:
save a playlist as a user scope playlist
play it with [p]playlist start playlistname
playlistname is not matched to any playlist
playlist id is also not matched

Playlist searcher is not seeming to pop up for loading a playlist. Have multiple playlists named the same thing and it's not asking which one to use. Playlist dedupe and playlist remove work appropriately (search popup occurs).

Formatting breaks on <https://www.youtube.com/watch?v=ocDlOD1Hw9k> in user facing responses because of ** in track name

The following commands and/or command groups should not be available to plebs/users without mod-admin perms in the server:
audioset autoplay command group
audioset dc
audioset notify
audioset restrictions command group

"The path must not contain spaces." can be removed from audioset localpath

Used: [p]playlist copy ENM --from-guild 154659242253484033 --to-guild 133049272517001216
Bot response: You cannot use --to-server       <-not embedded, add period at end, format like other "exceptions"

Drapersniper committed:
Address Aika's review

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper commented:
# Done
- [p]queue cleanself
- [p]help Audio
- [p]audioset autoplay reset 
- [p]audioset countrycode
- No matching is present in playlist names across scopes.
- Playlist searcher is not seeming to pop up for loading a playlist
- "The path must not contain spaces." can be removed from audioset localpath
- Formatting breaks

# Questions
- You cannot use --to-server <-not embedded, add period at end, format like other "exceptions"
  - This is not handled by Audio, its an early exit in the ArgParser which raises a BadArgument thats handled by the bot Event Handler, should we wish to format this we would need to let Audio handle these which would end up over complicating the core around this area (Errors handled by the Bot Event Handler do not have embed perms like Audio is allowed to enforce).
```py
commands.BadArgument("You cannot use `--guild`")
```
- The following commands and/or command groups should not be available to plebs/users without mod-admin perms in the server:
  - This is already locked behind a check
```py
    @checks.mod_or_permissions(manage_messages=True)
```

Drapersniper committed:
You know ... formatting sucks

Fixed formatting specially on queue command

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Add alias to `[p]audioset settings`

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Lets finally gracefully handle:
https://ptb.discordapp.com/channels/133049272517001216/387398816317440000/693322281489006653

Lets add some max_concurrency to audio commands

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
missed 1

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Casing

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Casing

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fix Type Hinting import

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
change required perms from manage_messages to manage_guild

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Cache spotify and youtube api keys and make use of `on_red_api_tokens_update`

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Forgot to black

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
small tweaks and fixes

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper commented on redbot/cogs/audio/core/utilities/formatting.py:
```diff
@@ -62,7 +65,10 @@ class FormattingUtilities(MixinMeta, metaclass=CompositeMetaClass):
-                name = f"**[{entry.get('name')}]({entry.get('url')})** - {str(entry.get('tracks')) + ' ' + _('tracks')}"
+                name = (
+                    f"**[{entry.get('name')}]({entry.get('url')})**"
+                    f" - {str(entry.get('tracks')) + ' ' + _('tracks')}"
```
```suggestion
                    " - {} {}".format(str(entry.get('tracks')),  _('tracks'))

```

Drapersniper committed:
small tweaks and fixes

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
revert max_concurrency

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
fix #3712 after discussing with aika

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
fu black

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
lets make Audio Pretty again

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
nothing to see here

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Better UX

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
`TimeoutError` handling

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Delete old data in chuncks

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
black y u do dis

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
*Sigh*

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Merge remote-tracking branch 'upstream/V3/develop' into C3-feature-audio-fixes

# Conflicts:
#	redbot/cogs/audio/manager.py

Drapersniper committed:
Fix ComplexScopeParser

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fix ComplexScopeParser

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Fix ComplexScopeParser

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Address the following:

Audioset autoplay reset
"Set auto-play playlist to play recently played tracks." - no cache enabled, where are these songs coming from... still the US popular playlist? If so, it shouldn't indicate it's recently played tracks

When autoplay is enabled and playing there is no "prev" or next button on the now display, even though prev and skip commands work.
Using "now" at other times does not show a prev button either.

[2020-04-01 14:32:00] [DEBUG] red.cogs.Audio.cog.Utilities.player: Current requester is <lavalink.rest_api.Track object at 0x000000B6E63E0A30>

Local folder <foldername> is classified as a search in embed footer: 03:11 until start of search playback: starts at #1 in queue

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Address the following:

Audioset autoplay reset
"Set auto-play playlist to play recently played tracks." - no cache enabled, where are these songs coming from... still the US popular playlist? If so, it shouldn't indicate it's recently played tracks

When autoplay is enabled and playing there is no "prev" or next button on the now display, even though prev and skip commands work.
Using "now" at other times does not show a prev button either.

[2020-04-01 14:32:00] [DEBUG] red.cogs.Audio.cog.Utilities.player: Current requester is <lavalink.rest_api.Track object at 0x000000B6E63E0A30>

Local folder <foldername> is classified as a search in embed footer: 03:11 until start of search playback: starts at #1 in queue

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Address the following:

Audioset autoplay reset
"Set auto-play playlist to play recently played tracks." - no cache enabled, where are these songs coming from... still the US popular playlist? If so, it shouldn't indicate it's recently played tracks

When autoplay is enabled and playing there is no "prev" or next button on the now display, even though prev and skip commands work.
Using "now" at other times does not show a prev button either.

[2020-04-01 14:32:00] [DEBUG] red.cogs.Audio.cog.Utilities.player: Current requester is <lavalink.rest_api.Track object at 0x000000B6E63E0A30>

Local folder <foldername> is classified as a search in embed footer: 03:11 until start of search playback: starts at #1 in queue

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
missed this

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
missed this

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
missed this

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Address:

Localtrack song titles show `\\` instead of `\` when using audioset status

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Address:

Localtrack song titles show `\\` instead of `\` when using audioset status

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Address:

Localtrack song titles show `\\` instead of `\` when using audioset status

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
...

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
...

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
...

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
improve random

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
improve random

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
improve random

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Resolved merge conflict

Drapersniper committed:
Resolved merge conflict

Drapersniper committed:
Merge branch 'C3-feature-audio-fixes' of https://github.com/Drapersniper/Red-DiscordBot into HEAD

Drapersniper committed:
Merge branch 'C3-feature-audio-fixes' of https://github.com/Drapersniper/Red-DiscordBot into HEAD

Drapersniper committed:
Make use of AsyncIter

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Make use of AsyncIter

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Merge branch 'C3-feature-audio-fixes' of https://github.com/Drapersniper/Red-DiscordBot into C3-feature-audio-fixes

Drapersniper committed:
fixes aikas crash + remove ":"

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
aaaaaaaaaarrrrrrrrrhhhhhhhhjjjjjjjjjjjj

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
black + remove global

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Merge remote-tracking branch 'upstream/V3/develop' into C3-feature-audio-fixes

Drapersniper committed:
Fixed `Loading track xx/xx.` looking like it was stuck and fix missing AsyncIter in playlists.py

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Revert black fuckery

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
fix aikas 2 crashes

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
remove dev tag from the build

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
black formatting

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
remove change log folder

Signed-off-by: Drapersniper <27962761+drapersniper@users.noreply.github.com>

Drapersniper committed:
Delete local_tracks.py

Drapersniper committed:
Apply suggestions from code review

Co-authored-by: jack1142 <6032823+jack1142@users.noreply.github.com>

Drapersniper committed:
Apply suggestions from code review

Co-authored-by: jack1142 <6032823+jack1142@users.noreply.github.com>

Drapersniper committed:
Update redbot/cogs/audio/core/utilities/queue.py

Drapersniper committed:
Update redbot/cogs/audio/core/utilities/playlists.py

Co-authored-by: jack1142 <6032823+jack1142@users.noreply.github.com>

Drapersniper committed:
Update redbot/cogs/audio/converters.py

Co-authored-by: jack1142 <6032823+jack1142@users.noreply.github.com>

Drapersniper committed:
Update redbot/cogs/audio/converters.py

Co-authored-by: jack1142 <6032823+jack1142@users.noreply.github.com>

Jackenmen committed:
remaining i18n stuff

Jackenmen committed:
suggestion 1

Jackenmen committed:
suggestion 2

Jackenmen committed:
suggestion 3

Jackenmen committed:
Revert "suggestion 1"

This reverts commit c0bd5eece6c3eb4eaa3ddec0d58ff95f95220bb7.

Drapersniper committed:
Remaining suggestions for `C3-feature-audio-fixes` (#6)

* remaining i18n stuff

* suggestion 1

* suggestion 2

* suggestion 3

Drapersniper committed:
Update redbot/cogs/audio/apis/__init__.py
